grakn-client/src/main/java/ai/grakn/client/QueryResponse.java

```diff
@@ -48,7 +50,13 @@ public Json getJsonResponse() {
         List<Json> json = Json.read(response).asJsonList();
         ArrayList<QueryResponse> result = new ArrayList<>();
         for (int i = 0; i < queries.size(); i++) {
-            result.add(new QueryResponse(queries.get(i), json.get(i)));
+            QueryResponse resp;
+            if(EMPTY_RESPONSE.equals(response)){
+                resp = new QueryResponse(queries.get(i), Json.nil());
+            } else {
+                resp = new QueryResponse(queries.get(i), json.get(i));
+            }
+            result.add(resp);
```

Comment:
This is in a loop but the `if` condition doesn't seem to depend on the loop at all. Why did the old code fail?

Reply:
because `json` in certain cases would only contain a single entry so when it tried to iterate over all entries with `i` it would get an index out of bounds exception.

Reply:
I am an idiot. I can clean this up MUCH more. Hold on . . .